kornia/augmentation/container/augment.py

```diff
@@ -368,74 +340,72 @@ def forward(  # type: ignore[override]
-                    pass  # Do nothing
-                elif isinstance(module, ImageSequential) and module.is_intensity_only() and dcate in DataKey:
-                    pass  # Do nothing
-                elif isinstance(module, VideoSequential) and dcate not in [DataKey.INPUT, DataKey.MASK]:
-                    batch_size: int = input.size(0)
-                    input = input.view(-1, *input.shape[2:])
-                    input, label = ApplyInverse.apply_by_key(input, label, module, param, dcate, extra_args=extra_args)
-                    input = input.view(batch_size, -1, *input.shape[1:])
-                elif isinstance(module, PatchSequential):
-                    raise NotImplementedError("Geometric involved PatchSequential is not supported.")
-                elif (
-                    isinstance(module, (GeometricAugmentationBase2D, ImageSequential, RandomErasing))
-                    and dcate in DataKey
-                ):
-                    input, label = ApplyInverse.apply_by_key(input, label, module, param, dcate, extra_args=extra_args)
-                elif isinstance(module, MixAugmentationBaseV2):
-                    if dcate in [DataKey.BBOX_XYXY, DataKey.BBOX_XYWH]:
-                        dcate = DataKey.BBOX
-                    input = module(input, params=param.data, data_keys=[dcate])
-                elif isinstance(module, (SequentialBase,)):
-                    raise ValueError(f"Unsupported Sequential {module}.")
-                else:
-                    raise NotImplementedError(f"data_key {dcate} is not implemented for {module}.")
+        outputs: Union[Tensor, List[DataType]] = in_args
+        for param in params:
+            module = self.get_submodule(param.name)
+            outputs = self.transform_op.transform(*outputs, module=module, param=param, extra_args=self.extra_args)
+            if not isinstance(outputs, (list, tuple)):
+                # Make sure we are unpacking a list whilst post-proc
+                outputs = [outputs]
+            self._update_transform_matrix_by_module(module)
+
+        outputs = self._arguments_postproc(args, outputs, data_keys=self.transform_op.data_keys)
+        # Restore it back
+        self.transform_op.data_keys = self.data_keys
+
+        self._params = params
+
+        if len(outputs) == 1 and isinstance(outputs, list):
+            return outputs[0]
+
+        return outputs
+
+    def _preproc_boxes(self, arg: DataType, dcate: DataKey) -> Boxes:
+        if DataKey.get(dcate) in [DataKey.BBOX]:
+            mode = "vertices_plus"
+        elif DataKey.get(dcate) in [DataKey.BBOX_XYXY]:
+            mode = "xyxy_plus"
+        elif DataKey.get(dcate) in [DataKey.BBOX_XYWH]:
+            mode = "xywh"
+        else:
+            raise ValueError(f"Unsupported mode `{DataKey.get(dcate).name}`.")
+        if isinstance(arg, (Boxes,)):
+            return arg
+        elif self.contains_video_sequential:
+            arg = cast(Tensor, arg)
```

Comment:
Can we do other thing here instead of just cast for the type checker? to have a runtime checker or proper unpack